code/__HELPERS/path.dm

```diff
@@ -400,35 +432,27 @@
 		//	if(destination_turf.density)
 		//		return TRUE
 		if(TURF_PATHING_PASS_PROC)
-			if(!destination_turf.CanAStarPass(ID, actual_dir , caller, no_id = no_id))
+			if(!destination_turf.CanAStarPass(ID, actual_dir, caller, no_id))
 				return TRUE
 		if(TURF_PATHING_PASS_NO)
 			return TRUE
 
+	var/static/list/directional_blocker_cache = typecacheof(list(/obj/structure/window, /obj/machinery/door/window, /obj/structure/railing, /obj/machinery/door/firedoor/border_only))
 	// Source border object checks
-	for(var/obj/structure/window/iter_window in src)
-		if(!iter_window.CanAStarPass(ID, actual_dir, no_id = no_id))
-			return TRUE
-
-	for(var/obj/machinery/door/window/iter_windoor in src)
-		if(!iter_windoor.CanAStarPass(ID, actual_dir, no_id = no_id))
-			return TRUE
-
-	for(var/obj/structure/railing/iter_rail in src)
-		if(!iter_rail.CanAStarPass(ID, actual_dir, no_id = no_id))
-			return TRUE
-
-	for(var/obj/machinery/door/firedoor/border_only/firedoor in src)
-		if(!firedoor.CanAStarPass(ID, actual_dir, no_id = no_id))
+	for(var/obj/lad in src)
```

Comment:
why?

Reply:
no typecheck good.

Reply:
the typecheck makes it faster though, since I'll have to filter those out anyway, or add an additional check for them.
Especially because objects are more common then mobs, significantly